auth/jwt/tokenizer.go

```diff
@@ -184,31 +192,24 @@ func parseOAuthToken(ctx context.Context, provider oauth2.Provider, token jwt.To
 		if !ok {
 			return auth.Key{}, errors.Wrap(ErrParseToken, fmt.Errorf("invalid claims for %s token", provider.Name()))
 		}
-		accessToken, ok := claims[oauthAccessTokenField].(string)
-		if !ok {
-			return auth.Key{}, errors.Wrap(ErrParseToken, fmt.Errorf("invalid access token claim for %s token", provider.Name()))
-		}
-		refreshToken, ok := claims[oauthRefreshTokenField].(string)
-		if !ok {
-			return auth.Key{}, errors.Wrap(ErrParseToken, fmt.Errorf("invalid refresh token claim for %s token", provider.Name()))
-		}
+
+		// access token and refresh token are not mandatory either of them can be present
+		accessToken, _ := claims[oauthAccessTokenField].(string)
+		refreshToken, _ := claims[oauthRefreshTokenField].(string)
```

Comment:
Fix according to the previous comment. Use only a single token (access or refresh):
```go
if ok {
		claims, ok := oauthToken.(pair)
		if !ok {
			return auth.Key{}, errors.Wrap(ErrParseToken, fmt.Errorf("invalid claims for %s token", provider.Name()))
		}

		// // access token and refresh token are not mandatory either of them can be present
		// accessToken, _ := claims[oauthAccessTokenField].(string)
		// refreshToken, _ := claims[oauthRefreshTokenField].(string)\
		switch claims.key {
		case oauthAccessTokenField:
			if err := provider.Validate(ctx, claims.value)
```
Also, fix the messy switch, since both cases do the same thing. Validation should inspect the error, since fallback to refresh may not make sense in some scenarios (revoked access, for example).